sympy/geometry/hyperbola.py

```diff
@@ -0,0 +1,986 @@
+
+    GeometryError
+        When `hradius`, `vradius` and `eccentricity` are incorrectly supplied
+        as parameters.
+    TypeError
+        When `center` is not a Point.
+
+    See Also
+    ========
+
+    Ellipse
+
+    Notes
+    -----
+    Constructed from a center and two radii, the first being the horizontal
+    radius (along the x-axis) and the second being the vertical radius (along
+    the y-axis).
+
+    When symbolic value for hradius and vradius are used, any calculation that
+    refers to the foci or the major or minor axis will assume that the hyperbola
+    has its major radius on the x-axis. If this is not true then a manual
+    rotation is necessary.
+
+    Examples
+    ========
+
+    >>> from sympy import Hyperbola, Point
+    >>> h1 = Hyperbola(Point(0, 0), 5, 3)
+    >>> h1.hradius, h1.vradius
+    (5, 3)
+    >>> h2 = Hyperbola(Point(3, 1), hradius=3, eccentricity= 2)
+    >>> h2
+    Hyperbola(Point2D(3, 1), 3, 3*sqrt(3))
+
+    """
+
+    def __new__(
+        cls, center=None, hradius=None, vradius=None, eccentricity=None, **kwargs):
+        hradius = sympify(hradius)
+        vradius = sympify(vradius)
+
+        eccentricity = sympify(eccentricity)
+
+        if center is None:
+            center = Point(0, 0)
+        else:
+            center = Point(center, dim=2)
+
+        if len(center) != 2:
+            raise ValueError('The center of "{}" must be a two dimensional point'.format(cls))
+
+        if len(list(filter(lambda x: x is not None, (hradius, vradius, eccentricity)))) != 2:
+            raise ValueError(filldedent('''
+                Exactly two arguments of "hradius", "vradius", and
+                "eccentricity" must not be None.'''))
+
+        if eccentricity is not None:
+            if eccentricity <= 1:
+                raise GeometryError("Eccentricity of hyperbola is always greater than 1")
+            elif hradius is None:
+                hradius = vradius / sqrt(eccentricity**2 - 1)
+            elif vradius is None:
+                vradius = hradius * sqrt(eccentricity**2 - 1)
+
+        #if hradius == vradius:
+        #    TODO: Implement Rectangular Hyperbola class for representing standard Rectangular Hyperbola Eq(x*y - c**2, 0)
+        #    return Rectangular_Hyperbola(center, hradius, **kwargs)
+
+        if S.Zero in (hradius, vradius):
+            return Segment(Point(center[0] - hradius, center[1] - vradius), Point(center[0] + hradius, center[1] + vradius))
+
+        if hradius.is_real is False or vradius.is_real is False:
+            raise GeometryError("Invalid value encountered when computing hradius / vradius.")
+
+        return GeometryEntity.__new__(cls, center, hradius, vradius, **kwargs)
+
+    def __contains__(self, o):
+        if isinstance(o, Point):
+            x = Dummy('x', real=True)
+            y = Dummy('y', real=True)
+
+            res = self.equation(x, y).subs({x: o.x, y: o.y})
+            return trigsimp(simplify(res)).is_zero
```

Comment:
It is possible to use something less computationally demanding here? Like cancel? 

Also, is trigsimp really needed? Is the idea that the point is described using trigonometric functions?

Reply:
Yeah true , I used `ellipse.py` as an inspiration for this part but giving a thought , I realize that trigsimp isn't really required !